### ISSUE TITLE
Using rx.angular.js with the Observer as syntax

### ISSUE THREAD

**Issue by PhiLhoSoft** [Reactive-Extensions/rx.angular.js#103]
I have a plunker to play with rx.angular.js: http://plnkr.co/edit/sGx4HH
If I replace `ng-controller="CentralController"` with `ng-controller="CentralController as c"` and add a c prefix to the scope variables, then use `observeOnScope($scope, 'vm.userInput')` and assign to `vm.results` instead of `$scope.results`, it just stops working!

Does rx.angular.js support this syntax? What other changes do I need to do?

**Comment by mattpodwysocki** [comment]
@PhiLhoSoft whatever the $scope.watch(expression) supports I suppose, we're doing nothing magical https://github.com/Reactive-Extensions/rx.angular.js/blob/master/src/observeonscope.js#L35

**Comment by mattpodwysocki** [comment]
@PhiLhoSoft apologizes, refactoring caused errors that somehow the unit tests weren't catching for some strange reason.

**Comment by PhiLhoSoft** [comment]
OK, perhaps I misunderstood the controller as syntax.
I had to do `observeOnScope($scope, function () { return vm.userInput; })` to have it to work.

**Comment by PhiLhoSoft** [comment]
For the record, it should have been `observeOnScope($scope, 'c.userInput')` ie. I should have used the name given in the view. Which I dislike, as it makes a dependency of the controller on the view. So I kept the function version.

Another problem: `$createObservableFunction` creates the function on $scope. So we cannot use the controllerAs syntax in the view. But we can refer to the undecorated function name.
Ie. with `"FooController as foo"`, we have to do `ng-change="doStuff()"` instead of `ng-change="foo.doStuff()", when we do `'$scope.$createObservableFunction((doStuff')`.

FYI.

**Comment by mattpodwysocki** [comment]
@PhiLhoSoft what would you prefer it to be? I'd love to hear recommendations.

**Comment by PhiLhoSoft** [comment]
Honestly, I have no idea... I report these for the record, but it seems hard to fix.
Most of rx.angular.js is based on $scope, which is OK, but $scope seems out of fashion, partly because of Angular 2 (which seems to use RxJS!), partly because of scope conflicts in nested controllers (which I rarely use, if ever, actually).

I am still not sure how this paradigm works... I have seen the name defined in "as xxx" in the scope in a case (declaration in the view, as shown below), and absent in another case (our application, perhaps because I made the declaration in the state declarations of ui-router).
In short, it is confusing.
Beside, as I wrote, I prefer to avoid referring to the name used in the view / declaration.

Anyway, I managed to go quite far in an experiment: http://plnkr.co/edit/UqdyB2?p=info
FYI...

Perhaps a workaround would be to systematize the usage of a function to resolve the variables or where to put them. Eg. `$createObservableFunction` would accept a function called with the created function, allowing to assign it to `this` or `vm`.

**Comment by PhiLhoSoft** [comment]
Actually, the example https://github.com/Reactive-Extensions/rx.angular.js/blob/master/examples/%24createObservableFunctionAsController/ uses controllerAs, and just pass `this` where scope is expected. Perhaps this should be clarified in the doc... If it is always that simple! (I assumed the functions expected a real $scope object).

**Comment by PhiLhoSoft** [comment]
Ah, I know why I missed it. It isn't documented in https://github.com/Reactive-Extensions/rx.angular.js/tree/master/docs yet. Anyway, nice move, I will start to use it right now. It might be the way to go.
